components/ethernet/include/esp_eth.h

```diff
@@ -118,12 +123,14 @@ typedef struct {
  * @note   config can not be NULL,and phy chip must be suitable to phy init func.
  *
  * @param[in] config  mac init data.
+ * @param[in] dma_rx_buf  DMA-capable buffer of size DMA_RX_BUF_SIZE * DMA_RX_BUF_NUM
+ * @param[in] dma_tx_buf  DMA-capable buffer of size DMA_TX_BUF_SIZE * DMA_TX_BUF_NUM
  *
  * @return
  *      - ESP_OK
  *      - ESP_FAIL
  */
-esp_err_t esp_eth_init(eth_config_t *config);
+esp_err_t esp_eth_init(eth_config_t *config, uint8_t *dma_rx_buf, uint8_t *dma_tx_buf);
```

Comment:
I suggest that we keep this function backwards compatible. To do so, could you please move these new options into `eth_config_t` structure?
In `esp_eth_init`, check if `dma_rx_buf` is NULL — in this case, allocate `CONFIG_DMA_RX_BUF_NUM` buffers. Same with TX buffers.

This way, applications which have previously zero-initialized `eth_config_t` structure will still work without any extra changes.